providers/implementations/rands/seeding/rand_cpu_x86.c

```diff
@@ -13,8 +13,14 @@
 #include "prov/seeding.h"
 
 #ifdef OPENSSL_RAND_SEED_RDCPU
+# if defined(OPENSSL_SYSNAME_TANDEM) && defined(_TNS_X_TARGET)
+#  include <builtin.h> /* _rdrand64 */
+#  include <string.h> /* memcpy */
+static size_t get_hardware_random_value(unsigned char *buf, size_t len);
```

Comment:
I'd prefer to see an approach where `get_hardware_random_value` always exists and calls the appropriate code on a per platform basis (i.e. `OPENSSL_ia32_rdseed_bytes`, `OPENSSL_ia32_rdrand_bytes` or `_rdrand64`.  This isolates the conditional code to one place and gives us a generic function that can be used elsewhere if required.  It also gives us cross architecture possibilities.

Reply:
That is actually a tiny bit closer to the original pull request, although I had everything in one procedure. I would be happy to further refactor this but I don't have a good basis of what platforms support the one of `OPENSSL_ia32_rdseed_bytes`, `OPENSSL_ia32_rdrand_bytes`. I can fold the discriminant into a commonly named procedure similar to the `get_hardware_random_value` for other platforms (better name?).

Reply:
They are generated by `gen_random` in crypto/x86cpuid.pl and c`rypto/x86_64cpuid.pl`.  They should exist for all x86 and x86_64 platforms.

It is weird that your platform isn't detecting them, I though the CPUID feature for x86/x86_64 was universal by the time these were introduced.

Reply:
I am surprised also, but the platform is not a traditional x86_64. It is Big Endian and has a whole bunch of strangeness. I have a related issue on the Itanium variant that I am working on for this also.